app/models/contribution.rb

```diff
@@ -121,6 +121,21 @@ def notify_to_backoffice(template_name, options = {})
     notify_once(template_name, financial_administrator, self, options) if financial_administrator
   end
 
+  def change_state_by_transaction(transaction_status)
+    case transaction_status
+      when 'waiting_payment' then
+        self.waiting
+      when 'paid', 'authorized' then
+        self.confirm
+      when 'pending_refund' then
+        self.request_refund
+      when 'refunded' then
+        self.refund
+      when 'refused' then
```

Comment:
acho q n, to tratando todos os casos de status retornado pelo pagarme justamente por serem diferentes dos que existem atualmente (que seguem o padrão do Moip se n me engano)